test/sandbox/test/integration_test.rb

```diff
@@ -769,4 +769,22 @@ def test_unsafe_postamble_component
       "Rendering UnsafePostambleComponent did not emit an HTML safety warning"
     )
   end
+
+  def test_renders_multiple_format_component_as_html
```

Comment:
Similar to [Rails default behavior](https://guides.rubyonrails.org/layouts_and_rendering.html#the-formats-option) do we want a test that validates we fall back to HTML in cases where the component is missing a format specific template?

Should we also have a test that validates that ViewComponent respects the `formats` option passed to `render` if we don't have one already?

Reply:
Hmm. My read of the docs is the opposite, that we should _not _ fall back for an unknown format:

> If a template with the specified format does not exist an ActionView::MissingTemplate error is raised.

Reply:
Also, I wrote a test for the `formats` option and realized that we don't pass _any_ of the render arguments into components: https://github.com/rails/rails/blob/4bb2227640531c877e30cc96f10df0c298dc3331/actionview/lib/action_view/template/renderable.rb#L16.

IMO we should look at passing render options to renderables, as it would be nice to handle format overrides.

Reply:
Ohhhh, you're right. I read that wrong. Good catch.

I was reading:

> Rails uses the format specified in the request (or :html by default)

Which I think means instead of passing `format = nil` in the compiler, it should be `format = :html`? 🤔

Reply:
I think we're passing `:html` by default in the compiler: https://github.com/ViewComponent/view_component/pull/2079/files#diff-fa28fa6e2d5f267384e7793b855281451a46b8437431867871a298fc52fe4940R316

I believe `request&.format&.to_sym` will also be `:html` by default, which gets passed to `render_template_for`.